external/vcpkg_custom_data/toolchains/linux_32.cmake

```diff
@@ -0,0 +1,572 @@
+    set(_VCPKG_TARGET_TRIPLET_PLAT freebsd)
+endif()
+
+set(VCPKG_TARGET_TRIPLET ${_VCPKG_TARGET_TRIPLET_ARCH}-${_VCPKG_TARGET_TRIPLET_PLAT} CACHE STRING "Vcpkg target triplet (ex. x86-windows)")
+set(_VCPKG_TOOLCHAIN_DIR ${CMAKE_CURRENT_LIST_DIR})
+
+if(NOT DEFINED _VCPKG_ROOT_DIR)
+    # Detect .vcpkg-root to figure VCPKG_ROOT_DIR
+    set(_VCPKG_ROOT_DIR_CANDIDATE ${CMAKE_CURRENT_LIST_DIR})
+    while(IS_DIRECTORY ${_VCPKG_ROOT_DIR_CANDIDATE} AND NOT EXISTS "${_VCPKG_ROOT_DIR_CANDIDATE}/.vcpkg-root")
+        get_filename_component(_VCPKG_ROOT_DIR_TEMP ${_VCPKG_ROOT_DIR_CANDIDATE} DIRECTORY)
+        if (_VCPKG_ROOT_DIR_TEMP STREQUAL _VCPKG_ROOT_DIR_CANDIDATE) # If unchanged, we have reached the root of the drive
+        else()
+            SET(_VCPKG_ROOT_DIR_CANDIDATE ${_VCPKG_ROOT_DIR_TEMP})
+        endif()
+    endwhile()
+    set(_VCPKG_ROOT_DIR ${_VCPKG_ROOT_DIR_CANDIDATE} CACHE INTERNAL "Vcpkg root directory")
+endif()
+
+_vcpkg_get_directory_name_of_file_above(_VCPKG_ROOT_DIR ${CMAKE_CURRENT_LIST_DIR} ".vcpkg-root")
+if(NOT _VCPKG_ROOT_DIR)
+    message(FATAL_ERROR "Could not find .vcpkg-root")
+endif()
+
+if (NOT DEFINED _VCPKG_INSTALLED_DIR)
+    if(_VCPKG_MANIFEST_DIR)
+        set(_VCPKG_INSTALLED_DIR ${CMAKE_BINARY_DIR}/vcpkg_installed)
+    else()
+        set(_VCPKG_INSTALLED_DIR ${_VCPKG_ROOT_DIR}/installed)
+    endif()
+
+    set(_VCPKG_INSTALLED_DIR ${_VCPKG_INSTALLED_DIR}
+        CACHE PATH
+        "The directory which contains the installed libraries for each triplet")
+endif()
+
+if(CMAKE_BUILD_TYPE MATCHES "^[Dd][Ee][Bb][Uu][Gg]$" OR NOT DEFINED CMAKE_BUILD_TYPE) #Debug build: Put Debug paths before Release paths.
+    list(APPEND CMAKE_PREFIX_PATH
+        ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}
+    )
+    list(APPEND CMAKE_LIBRARY_PATH
+        ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib/manual-link ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib/manual-link
+    )
+    list(APPEND CMAKE_FIND_ROOT_PATH
+        ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}
+    )
+else() #Release build: Put Release paths before Debug paths. Debug Paths are required so that CMake generates correct info in autogenerated target files.
+    list(APPEND CMAKE_PREFIX_PATH
+        ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET} ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug
+    )
+    list(APPEND CMAKE_LIBRARY_PATH
+        ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib/manual-link ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib/manual-link
+    )
+    list(APPEND CMAKE_FIND_ROOT_PATH
+        ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET} ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug
+    )
+endif()
+
+# If one CMAKE_FIND_ROOT_PATH_MODE_* variables is set to ONLY, to  make sure that ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}
```

Comment:
nit: extra space here "ONlY, to  make sure..."

Reply:
This file is combination of settings required to build x86 in cmake (first 3 lines) and a verbatim copy of the vcpkg cmake toolchain file (required to locate packages built by vcpkg). This was one of the ways to chain 2 separate toolchains into one. Another way of doing this is by specifying another cmake option
```
cmake ../my/project \
   -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake \
   -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=../my/project/compiler-settings-toolchain.cmake
```
This adds to our list of things to remember for manual builds. 
Am going to resolve the comments in this file without making changes as it will be easier for us to diff things in the future.